packages/core/src/transport/httpRequest.spec.ts

```diff
@@ -19,62 +19,185 @@ describe('httpRequest', () => {
-    if (!navigator.sendBeacon) {
-      pending('no sendBeacon support')
-    }
+    it('should use sendBeacon when the bytes count is correct', () => {
+      if (!navigator.sendBeacon) {
+        pending('no sendBeacon support')
+      }
 
-    request.send('{"foo":"bar1"}\n{"foo":"bar2"}', 10)
+      request.send('{"foo":"bar1"}\n{"foo":"bar2"}', 10)
 
-    expect(requests.length).toEqual(1)
-    expect(requests[0].type).toBe('sendBeacon')
-  })
+      expect(requests.length).toEqual(1)
+      expect(requests[0].type).toBe('sendBeacon')
+    })
+
+    it('should use xhr over sendBeacon when the bytes count is too high', () => {
+      request.send('{"foo":"bar1"}\n{"foo":"bar2"}', BATCH_BYTES_LIMIT)
+
+      expect(requests.length).toEqual(1)
+      expect(requests[0].type).toBe('xhr')
+    })
+
+    it('should fallback to xhr when sendBeacon is not queued', () => {
+      if (!navigator.sendBeacon) {
+        pending('no sendBeacon support')
+      }
+      spyOn(navigator, 'sendBeacon').and.callFake(() => false)
+
+      request.send('{"foo":"bar1"}\n{"foo":"bar2"}', 10)
+
+      expect(requests.length).toEqual(1)
+      expect(requests[0].type).toBe('xhr')
+    })
 
-  it('should use xhr over sendBeacon when the bytes count is too high', () => {
-    request.send('{"foo":"bar1"}\n{"foo":"bar2"}', BATCH_BYTES_LIMIT)
+    it('should fallback to xhr when sendBeacon throws', () => {
+      if (!navigator.sendBeacon) {
+        pending('no sendBeacon support')
+      }
+      spyOn(navigator, 'sendBeacon').and.callFake(() => {
+        throw new TypeError()
+      })
 
-    expect(requests.length).toEqual(1)
-    expect(requests[0].type).toBe('xhr')
+      request.send('{"foo":"bar1"}\n{"foo":"bar2"}', 10)
+
+      expect(navigator.sendBeacon).toHaveBeenCalled()
+      expect(requests.length).toEqual(1)
+      expect(requests[0].type).toBe('xhr')
+    })
   })
 
-  it('should fallback to xhr when sendBeacon is not queued', () => {
-    if (!navigator.sendBeacon) {
-      pending('no sendBeacon support')
-    }
-    spyOn(navigator, 'sendBeacon').and.callFake(() => false)
+  describe('send (with fetch-keepalive FF)', () => {
+    beforeEach(() => {
+      updateExperimentalFeatures(['fetch-keepalive'])
+    })
+
+    afterEach(() => {
+      resetExperimentalFeatures()
+    })
 
-    request.send('{"foo":"bar1"}\n{"foo":"bar2"}', 10)
+    it('should use xhr when fetch keepalive is not available', () => {
+      const originalRequest = window.Request
+      ;(window.Request as any) = false
 
-    expect(requests.length).toEqual(1)
-    expect(requests[0].type).toBe('xhr')
+      request.send('{"foo":"bar1"}\n{"foo":"bar2"}', 10)
+
+      expect(requests.length).toEqual(1)
+      expect(requests[0].type).toBe('xhr')
+      expect(requests[0].url).toContain(ENDPOINT_URL)
+      expect(requests[0].body).toEqual('{"foo":"bar1"}\n{"foo":"bar2"}')
+      window.Request = originalRequest
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏restore the state in an `afterEach` callback